### ISSUE TITLE
Bad Landsat File

### ISSUE THREAD

**Issue by jessjaco** [microsoft/PlanetaryComputer#114]
I'm getting an error with the file at https://landsateuwest.blob.core.windows.net/landsat-c2/level-2/standard/oli-tirs/2021/055/076/LC08_L2SR_055076_20211231_20220106_02_T2/LC08_L2SR_055076_20211231_20220106_02_T2_SR_B4.TIF

```python
from planetary_computer import sign
import rioxarray

url = 'https://landsateuwest.blob.core.windows.net/landsat-c2/level-2/standard/oli-tirs/2021/055/076/LC08_L2SR_055076_20211231_20220106_02_T2/LC08_L2SR_055076_20211231_20220106_02_T2_SR_B4.TIF'
rioxarray.open_rasterio(sign(url)).rio.to_raster('test.tif')

ERROR 1: TIFFFillTile:Read error at row 6144, col 3840, tile 295; got 81147 bytes, expected 101991
ERROR 1: TIFFReadEncodedTile() failed.
ERROR 1: IReadBlock failed at X offset 25, Y offset 9: TIFFReadEncodedTile() failed.
Traceback (most recent call last):
  File "rasterio/_io.pyx", line 936, in rasterio._io.DatasetReaderBase._read
  File "rasterio/_io.pyx", line 174, in rasterio._io.io_multi_band
  File "rasterio/_err.pyx", line 191, in rasterio._err.exc_wrap_int
rasterio._err.CPLE_AppDefinedError: IReadBlock failed at X offset 25, Y offset 9: TIFFReadEncodedTile() failed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/rioxarray/raster_array.py", line 1137, in to_raster
    return RasterioWriter(raster_path=raster_path).to_raster(
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/rioxarray/raster_writer.py", line 271, in to_raster
    data = encode_cf_variable(out_data).values.astype(numpy_dtype)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/conventions.py", line 273, in encode_cf_variable
    var = coder.encode(var, name=name)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/coding/times.py", line 664, in encode
    dims, data, attrs, encoding = unpack_for_encoding(variable)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/coding/variables.py", line 104, in unpack_for_encoding
    return var.dims, var.data, var.attrs.copy(), var.encoding.copy()
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/dataarray.py", line 670, in data
    return self.variable.data
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/variable.py", line 354, in data
    return self.values
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/variable.py", line 527, in values
    return _as_array_or_item(self._data)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/variable.py", line 267, in _as_array_or_item
    data = np.asarray(data)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/indexing.py", line 653, in __array__
    self._ensure_cached()
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/indexing.py", line 650, in _ensure_cached
    self.array = NumpyIndexingAdapter(np.asarray(self.array))
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/indexing.py", line 623, in __array__
    return np.asarray(self.array, dtype=dtype)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/indexing.py", line 524, in __array__
    return np.asarray(array[self.key], dtype=None)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/rioxarray/_io.py", line 322, in __getitem__
    return indexing.explicit_indexing_adapter(
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/xarray/core/indexing.py", line 815, in explicit_indexing_adapter
    result = raw_indexing_method(raw_key.tuple)
  File "/srv/conda/envs/notebook/lib/python3.10/site-packages/rioxarray/_io.py", line 306, in _getitem
    out = riods.read(band_key, window=window, masked=self.masked)
  File "rasterio/_io.pyx", line 612, in rasterio._io.DatasetReaderBase.read
  File "rasterio/_io.pyx", line 939, in rasterio._io.DatasetReaderBase._read
rasterio.errors.RasterioIOError: Read or write failed. IReadBlock failed at X offset 25, Y offset 9: TIFFReadEncodedTile() failed.
```

**Comment by jessjaco** [comment]
Getting a similar (but slightly different) error with https://landsateuwest.blob.core.windows.net/landsat-c2/level-2/standard/oli-tirs/2021/096/061/LC08_L2SP_096061_20211215_20211223_02_T2/LC08_L2SP_096061_20211215_20211223_02_T2_SR_B5.TIF as well

**Comment by TomAugspurger** [comment]
Thanks. I confirmed the same error with the first URL. I manually pulled the same file from USGS (https://earthexplorer.usgs.gov/scene/metadata/full/5e83d14f2fc39685/LC80550762021365LGN00/ is the matching item) but didn't see an error. We'll mark that for reprocessing.

Tracking the second issue in https://github.com/microsoft/PlanetaryComputer/discussions/101. We'll be reprocessing those files soon hopefully.
